pennylane/tape/tapes/qubit_param_shift.py

```diff
@@ -138,23 +138,25 @@ def parameter_shift(self, idx, params, **options):
         op = self._par_info[t_idx]["op"]
         p_idx = self._par_info[t_idx]["p_idx"]
 
-        s = (
-            np.pi / 2
-            if op.grad_recipe is None or op.grad_recipe[p_idx] is None
-            else op.grad_recipe[p_idx]
-        )
-        s = options.get("shift", s)
+        recipe = op.grad_recipe[p_idx]
+        s = options.get("shift", np.pi / 2)
+        c = 1 / (2 * np.sin(s))

```

Comment:
@trbromley: currently, the ability to set the shift *only* works for gates where `grad_recipe=None`, in which case we default to the two-term parameter shift rule. Unfortunately, I can't quite tell how to extend it to the 4-term rule 😢 

We would have to:

* Pass a _list_ of shifts, one for each degree of freedom (e.g., alpha and beta for the 4-term shift)

* Solve the implicit equation to determine c1 and c2 :/

Reply:
Since in this case, we'd have the shifts provided by the user, it might be that we could provide values for `c1` and `c2` explicitly.

Had a chat with @josh146 on this and one thing to decide here would be to decide the correct UI to allow passing shifts to single-qubit and multi-qubit operations. Therefore, might indeed be worth revisiting later.